src/core/src/op/type_relaxed.cpp

```diff
@@ -35,9 +35,10 @@ void convert_types(std::shared_ptr<v0::Parameter>& parameter,
     auto& tensor = output.get_tensor();
 
     if (lower_success || upper_success) {
-        OPENVINO_SUPPRESS_DEPRECATED_START
-        tensor.set_element_type(new_type);  // deprecated piece
-        OPENVINO_SUPPRESS_DEPRECATED_END
+        if (auto parameter = dynamic_cast<ov::op::v0::Parameter*>(output.get_node())) {
+            parameter->set_element_type(new_type);
```

Comment:
it does not look like this `output` is actually connected to `Parameter`, or I'm missing something ?